### PR TITLE
14637 Launch draft restoration filing

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#14637

*Description of changes:*
- added Restore Company menu item and handling to StaffNotation component
- added Restoration Filing interface
- added buildRestorationFiling()
- misc cleanup
- updated/added unit tests
- app version = 3.9.3

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/FilingHistoryList.vue [hunk 1]
```diff
@@ -43,6 +36,13 @@
       </div>
     </v-fade-transition>
 
+    <!-- Court order notification -->
+    <v-card v-if="hasCourtOrders" class="my-6 pa-6" elevation="0">
+      <v-icon>mdi-gavel</v-icon>
+      <span class="ml-1">Court order(s) have been filed on this company. Review the filing history for impacts
+        to business information.</span>
+    </v-card>
+
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:45]: I moved this down to be consistent with other components where we first have dialogs, then the spinner, then the page components.

File: src/components/Dashboard/StaffNotation.vue [hunk 2]
```diff
@@ -25,14 +25,6 @@
       courtOrderNumberRequired="true"
     />
 
-    <AddStaffNotationDialog
-      :dialog="isAddingPutBackOn"
-      @close="hidePutBackOnDialog($event)"
-      attach="#staff-notation"
-      :displayName="FilingNames.PUT_BACK_ON"
-      :name="FilingTypes.PUT_BACK_ON"
-    />
```
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:34]: I moved this and other blocks to be in the same order as in the drop-down menu.

File: src/components/Dashboard/StaffNotation.vue [hunk 3]
```diff
@@ -55,37 +55,76 @@
 
         <v-list dense>
           <v-list-item-group color="primary">
-            <v-list-item @click="showRegistrarsNotationDialog()" :disabled="disabled">
+            <v-list-item
+              data-type="registrars-notation"
+              @click="showRegistrarsNotationDialog()"
+              disabled="disabled"
+            >
```
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:62]: I added `data-type` attributes to all items for easier testing.

File: src/components/Dashboard/StaffNotation.vue [hunk 4]
```diff
@@ -184,14 +221,42 @@ export default class StaffNotation extends Vue {
...
+      // navigate to Create UI
+      const url = `${this.createUrl}?id=${this.getIdentifier}`
+      navigate(url)
+    } catch (error) {
+      // clear spinner on error
+      this.$root.$emit('showSpinner', false)
+
+      alert(`Could not create restoration filing. Please try again or cancel.`)
+    }
+  }
```
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:246]: The method above is new. The methods below were simply moved.

File: src/components/Dashboard/TodoList.vue [hunk 5]
```diff
@@ -470,8 +470,7 @@ import { ActionBindingIF, ApiTaskIF, BusinessIF, BusinessWarningIF, ConfirmDialo
   mixins: [
     AllowableActionsMixin,
     DateMixin,
-    EnumMixin,
-    FilingMixin
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:474]: Not used in this component.

File: src/interfaces/restoration-interfaces.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * A filing's restoration object from the API. See:
+ * https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/restoration.json
+ */
+export interface RestorationIF {
+  date: string // FUTURE: describe format here
+  type: RestorationTypes
+  expiry?: string // FUTURE: describe format here
+}
```
Comment (severinbeauvais) [inline src/interfaces/restoration-interfaces.ts:11]: This is from src/interfaces/restoration-interface.ts, which I renamed.

File: src/interfaces/restoration-interfaces.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import { RestorationTypes } from '@/enums'
```
Comment (severinbeauvais) [inline src/interfaces/restoration-interfaces.ts:1]: Actually I renamed src/interfaces/restoration-interface.ts -> this.

File: src/store/getters.ts [hunk 8]
```diff
@@ -48,8 +48,7 @@ export default {
   },
 
   /** Is True if there is a blocker including firm compliance. */
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  hasBlocker (state: StateIF, getters: any): boolean {
+  hasBlocker (_state: StateIF, getters: any): boolean {
```
Comment (severinbeauvais) [inline src/store/getters.ts:51]: A little syntax thing: if you name your formal parameters starting with a `_`, eslint ignores them.

**Comment by severinbeauvais** [comment]
@argush3 If you're working on the Legal API accepting a PUT/POST restoration filing, you can checkout this code which _should_ work (though I'm still working on unit tests).

File: src/components/Dashboard/StaffNotation.vue [hunk 9]
```diff
@@ -166,13 +212,9 @@ export default class StaffNotation extends Vue {
...
-    this.isAddingPutBackOn = true
-  }
-
-  hidePutBackOnDialog (needReload: boolean): void {
-    this.isAddingPutBackOn = false
-    this.close(needReload)
+  goToConversionFiling ():void {
+    const url = `${this.editUrl}${this.getIdentifier}/conversion`
+    navigate(url)
   }
```
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:218]: Was moved up.

Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:218]: Were moved down.

Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:218]: goToConversionFiling() was moved up.

**Review by ChrisGabel23 — APPROVED** [review]

**Pull request merged**